rest/src/main/java/com/nikolay/rest/errorHandler/RestErrorHandler.java

```diff
@@ -66,8 +75,10 @@
     public ResponseEntity<Object> handleEmptyResultDataAccessException(
             EmptyResultDataAccessException ex, WebRequest request) {
         LOGGER.error("handleEmptyResultDataAccessException() message: " + ex.getLocalizedMessage());
-        ApiError apiError = new ApiError(HttpStatus.BAD_REQUEST, ex.getLocalizedMessage());
-        return handleExceptionInternal(ex, apiError, new HttpHeaders(), HttpStatus.BAD_REQUEST,
+        ApiError apiError = new ApiError(HttpStatus.INTERNAL_SERVER_ERROR,
```

Comment:
please explain, why is this http status INTERNAL_SERVER_ERROR?

Reply:
DataAccessException occurs on the server when, for example, it cannot save to the database.
And Internal Server status code indicates that the serve encountered an unexpected condition that prevented it from fulfilling the request.